packages/kcmsf/src/pages/match.tsx

```diff
@@ -10,27 +10,90 @@ import { Judge } from "../utils/match/judge";
 import { expiryTimestamp, parseSeconds } from "../utils/time";
 
 type TimerState = "Initial" | "Started" | "Finished";
-
+type TeamFetchRes = {
+  id: string;
+  name: string;
+  entryId: string;
+  members: string[2];
+  clubName: string;
+  robotType: RobotType;
+  category: "elementary" | "open";
+  isEntered: boolean;
+};
 export const Match = () => {
-  const matchInfo = useLocation().state as MatchInfo | undefined;
-  const isExhibition = matchInfo == null;
+  const { id, matchType } = useParams<{ id: string; matchType: MatchType }>();
+  const isExhibition = id == null || matchType == null;
+  const [matchInfo, setMatchInfo] = useState<MatchInfo | undefined>(undefined);
+  const [matchJudge, setMatchJudge] = useState<Judge>(
+    new Judge(
+      { multiWalk: false },
+      { multiWalk: false },
+      { matchInfo },
+      { matchInfo }
+    )
+  );
```

Comment:
L78で再生成するなら初期値はいらないように思われます。
```suggestion
  const [matchJudge, setMatchJudge] = useState<Judge>();
```